CONTRIBUTING.md

```diff
@@ -49,3 +49,19 @@ This project follows
 
 We pledge to maintain an open and welcoming environment. For details, see our
 [code of conduct](https://dart.dev/code-of-conduct).
+
+## Tests
+
+Packages `native_assets_cli`, `native_assets_builder`, and `native_toolchain_c`
+roll into the Dart SDK and flutter_tools. The tests of these packages are run
+on the Dart SDK in a different way than on the GitHub actions on this repo.
```

Comment:
Maybe link to an issue in the Dart SDK - this is not great to have this schism, it should be fixed. Maybe with the workspace feature migration @jakemac53 was working on.

Reply:
I don't believe there's a tracking issue. I've filed https://github.com/dart-lang/sdk/issues/56574.

I'd love to reduce the diff between what's used outside the Dart SDK (`package:test`) and inside the Dart SDK (`tools/test.py` and `package:test_runner`), but I don't know how much buy in there is to reduce this tech debt.

cc @johnmccutchan It's not Flutter that has multiple ways of running tests, also Dart. Here's some ammunition for our quest!